File: 프로그래머스/2/49994. 방문 길이/방문 길이.py
def solution(dirs):
    dx = [0,1,0,-1]
    dy = [1,0,-1,0]
    answer = 0
    visited = []
    x, y = 5, 5
    
    for d in dirs:
        if d == 'L':
            nx, ny = x + dx[2], y + dy[2]
        if d == 'U':
            nx, ny = x + dx[3], y + dy[3]
        if d == 'R':
            nx, ny = x + dx[0], y + dy[0]
        if d == 'D':
            nx, ny = x + dx[1], y + dy[1]
        
        if 0<=nx<11 and 0<=ny<11:
            if [x,y,nx,ny] not in visited and [nx,ny,x,y] not in visited:
                answer+=1
                visited.append([x,y,nx,ny])
            x, y = nx, ny    
        
    return answer

File: 프로그래머스/2/49994. 방문 길이/방문 길이.js
function solution(dirs) {
    const move = { U: [0, 1], D: [0, -1], R: [1, 0], L: [-1, 0] };
    let cur = [0, 0];
    const moveList = new Set();
    
    for (const dir of dirs) {
        const nextX = cur[0] + move[dir][0];
        const nextY = cur[1] + move[dir][1];
        
        if (nextX > 5 || nextX < -5 || nextY > 5 || nextY < -5) continue;
        
        moveList.add(`${cur[0]}${cur[1]}${nextX}${nextY}`);
        moveList.add(`${nextX}${nextY}${cur[0]}${cur[1]}`);
        
        cur = [nextX, nextY];
    }
    
    return moveList.size / 2;
}